services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java

```diff
@@ -30,54 +32,93 @@
 import software.amazon.awssdk.transfer.s3.S3ClientConfiguration;
 import software.amazon.awssdk.transfer.s3.S3TransferManager;
 import software.amazon.awssdk.transfer.s3.Upload;
+import software.amazon.awssdk.transfer.s3.UploadDirectory;
+import software.amazon.awssdk.transfer.s3.UploadDirectoryRequest;
 import software.amazon.awssdk.transfer.s3.UploadRequest;
 import software.amazon.awssdk.utils.CompletableFutureUtils;
 
 @SdkInternalApi
 public final class DefaultS3TransferManager implements S3TransferManager {
     private final S3CrtAsyncClient s3CrtAsyncClient;
+    private final TransferConfiguration transferConfiguration;
+    private final UploadDirectoryManager uploadDirectoryManager;
 
     public DefaultS3TransferManager(DefaultBuilder tmBuilder) {
         S3CrtAsyncClient.S3CrtAsyncClientBuilder clientBuilder = S3CrtAsyncClient.builder();
+        TransferConfiguration.Builder transferConfigBuilder = TransferConfiguration.builder();
         if (tmBuilder.s3ClientConfiguration != null) {
             tmBuilder.s3ClientConfiguration.credentialsProvider().ifPresent(clientBuilder::credentialsProvider);
             tmBuilder.s3ClientConfiguration.maxConcurrency().ifPresent(clientBuilder::maxConcurrency);
             tmBuilder.s3ClientConfiguration.minimumPartSizeInBytes().ifPresent(clientBuilder::minimumPartSizeInBytes);
             tmBuilder.s3ClientConfiguration.region().ifPresent(clientBuilder::region);
             tmBuilder.s3ClientConfiguration.targetThroughputInGbps().ifPresent(clientBuilder::targetThroughputInGbps);
             tmBuilder.s3ClientConfiguration.asyncConfiguration().ifPresent(clientBuilder::asyncConfiguration);
+
+            tmBuilder.s3ClientConfiguration.uploadDirectoryConfiguration().ifPresent(transferConfigBuilder::configuration);
         }
 
         s3CrtAsyncClient = clientBuilder.build();
+        transferConfiguration = transferConfigBuilder.build();
+        uploadDirectoryManager = new UploadDirectoryManager(transferConfiguration, this::upload);
     }
 
     @SdkTestInternalApi
-    DefaultS3TransferManager(S3CrtAsyncClient s3CrtAsyncClient) {
+    DefaultS3TransferManager(S3CrtAsyncClient s3CrtAsyncClient, UploadDirectoryManager uploadDirectoryManager) {
```

Comment:
Why does this need to be exposed for testing purposes? Can't we get to this via the builder?